ui/pages/token-details/token-details-page.js

```diff
@@ -0,0 +1,229 @@
+              data-testid="token-details-page__token-contract-address"
+            >
+              {token.address}
+            </Typography>
+            <Tooltip
+              position="bottom"
+              title={copied ? t('copiedExclamation') : t('copyToClipboard')}
+              containerClassName="token-details__copy-icon"
+            >
+              <button
+                type="link"
+                onClick={() => {
+                  handleCopy(token.address);
+                }}
+                title=""
+                className="token-details__copyButton"
+                data-testid="token-details-page__copy-button"
+              >
+                <CopyIcon size={11} color="#037DD6" />
+              </button>
+            </Tooltip>
+          </Box>
+          <Typography
+            variant={TYPOGRAPHY.H9}
+            margin={[4, 0, 0, 0]}
+            color={COLORS.UI4}
+            fontWeight={FONT_WEIGHT.BOLD}
+            data-testid="token-details-page__token-decimal-title"
+          >
+            {t('tokenDecimalTitle')}
+          </Typography>
+          <Typography
+            variant={TYPOGRAPHY.H7}
+            margin={[1, 0, 0, 0]}
+            color={COLORS.BLACK}
+            data-testid="token-details-page__token-decimals"
+          >
+            {token.decimals}
+          </Typography>
+          <Typography
+            variant={TYPOGRAPHY.H9}
+            margin={[4, 0, 0, 0]}
+            color={COLORS.UI4}
+            fontWeight={FONT_WEIGHT.BOLD}
+            data-testid="token-details-page__current-network-title"
+          >
+            {t('network')}
+          </Typography>
+          <Typography
+            variant={TYPOGRAPHY.H7}
+            margin={[1, 0, 0, 0]}
+            color={COLORS.BLACK}
+            data-testid="token-details-page__current-network"
+          >
+            {networkType === NETWORK_TYPE_RPC
+              ? networkNickname ?? t('privateNetwork')
+              : t(networkType)}
+          </Typography>
+          <Button
+            type="primary"
+            className="token-details__hide-token-button"
+            onClick={() => {
+              dispatch(
+                showModal({ name: 'HIDE_TOKEN_CONFIRMATION', token, history }),
+              );
+            }}
+            data-testid="token-details-page__hide-token-button"
+          >
+            <Typography
+              variant={TYPOGRAPHY.H6}
+              color={COLORS.PRIMARY1}
+              data-testid="token-details-page__hide-token"
+            >
+              {t('hideToken')}
+            </Typography>
+          </Button>
+        </Box>
+      </Box>
+    );
+  };
+
+  const renderTokenDetailsPage = () => {
+    let content;
+
+    if (token) {
+      content = renderTokenDetailsComponent();
+    } else {
+      content = <Redirect to={{ pathname: DEFAULT_ROUTE }} />;
+    }
+
+    return content;
+  };
+
+  return renderTokenDetailsPage();
```

Comment:
before the main JSX of this component, check to see if token is defined. If it isn't,:
``` return <Redirect to={{ pathaname: DEFAULT_ROUTE }} />```

otherwise return the JSX. This is cleaner and easier to follow than the function assignments and calls, and might have some optimization wins as React will deal with the different JSX trees internally versus function memory assignment on every render.

Reply:
@brad-decker 
Fixed in: fc353df083ca4099424224735ae80831198a6dcd